### PR TITLE
Stabilize dynamic endpoints resolution

### PR DESCRIPTION
**Pull request by deldesir**
Construct the endpoint path no matter what the root url is. Smoke and functional test on Ubuntu 24.04 (Noble Numbat).

### CODE REVIEW

File: cps/editbooks.py [hunk 1]
```diff
@@ -332,7 +332,8 @@ def request_media_download(media_url, original_url):
     
     if request.method == "POST" and "mediaURL" in request.form:
         media_url = request.form["mediaURL"]
-        original_url = request.referrer + "meta"
+        original_url = request.form["serverURL"]
+        original_url = original_url.replace("/media", "/meta")
```
Comment (holta) [inline cps/editbooks.py:336]: Is this a REGEX that should possible be a bit safer, e.g. in cases where implementer chose http://box.lan/media in their school?  (e.g. to ensure http://box.lan/media/media is being changed to http://box.lan/media/meta ?)

```suggestion
        original_url = original_url.replace("/media$", "/meta")
```

Comment (deldesir) [reply cps/editbooks.py:336]: Thanks for mentioning this particular use case. Let's make sure that only occurrence of "/media" at the end of the URL path is replaced with "/meta".

**Comment by holta** [comment]
Quick Recap after code review:

1. This PR is being merged (to expand user testing) even though code will evolve after merging in coming days or hours.

2. @deldesir will look intro strengthening https://github.com/iiab/calibre-web/pull/33/files#r1392558784

3. `var endpoint = currentURL.href;` wasn't actually a problem, but was a bit confusing as I didn't realize this referred to an [API endpoint](https://smartbear.com/learn/performance-monitoring/api-endpoints/) rather than a network endpoint.

   So for now, @deldesir has reverted to using string `currentURL.href` directly: https://github.com/iiab/calibre-web/pull/33/commits/582db4e6381a2aebfbb66fcc06c1e4cd01b1b2b0

   RELATED:

   - https://developer.mozilla.org/en-US/docs/Web/API/URL/href (string) is essentially a superset of...
   - https://developer.mozilla.org/en-US/docs/Web/API/URL/pathname (string)

**Review by deldesir — COMMENTED** [review]
Should be more stable now

**Comment by holta** [comment]
> Should be more stable now

Interesting that Python [rsplit](https://www.w3schools.com/python/ref_string_rsplit.asp) returns a list of 2 items not just 1 item, ok...

```
>>> "http://box.lan/media/media".rsplit("/media", 1)
['http://box.lan/media', '']
>>> "http://box.lan/media/media".rsplit("/media", 1)[0]
'http://box.lan/media'
>>> "http://box.lan/media/media".rsplit("/media", 1)[0] + "/meta"
'http://box.lan/media/meta'
```

**Pull request merged**